### DISCUSSION TITLE
make 'drag to seek' diff from 'click to seek'

### DISCUSSION THREAD

**Discussion by hooke007** [tomasklaen/uosc#300] [General]
It could be a catastrophy of performance if users seek by dragging. The condition would became worse when combining it with thumbnail preview.

We may need to make 'drag to seek' diff from 'click to seek', just like what osc did.
https://github.com/mpv-player/mpv/blob/cc65b3892d89ae35d31067fba285ab20716a1aee/player/lua/osc.lua#L2027-L2041

- Current

Note: I am using a simplified thumbfast.lua which performs much better than original version.
Considering most users doesn't have the powerful CPU, they will feel obvious lagging.

https://user-images.githubusercontent.com/41094733/194409222-ac6eddaf-501b-4dff-8dee-f47e8ed3146b.mp4

- Try [`fastseek`](https://mpv.io/manual/master/#on-screen-controller-seekbarkeyframes)

https://user-images.githubusercontent.com/41094733/194409392-7249094b-2cc9-4581-ad4f-69b9d4216a6d.mp4

**Comment by tomasklaen** [comment]
I've tried it just today, and although this is fine on longer videos, drag seeking on shorter ones makes timeline look like it's having a seizure. The handle is jumping all over the place. This becomes even more of an issue on videos with big keyframe distances.

I'll add an option for it, but it'll be disabled by default.

**Reply by christoph-heinrich** [reply]
We could do keyframe seeks while dragging around, and then when the user stops (releases mouse button or stops moving for long enough) we do an exact seek. That way we don't need another option.

**Comment by christoph-heinrich** [comment]
> Note: I am using a simplified thumbfast.lua which performs much better than original version.

Please share your improvments with the rest of us by making a PR :pray:

**Reply by hooke007** [reply]
I do not think it will be received. For I cleaned all unnecessary (I think) process of encoding such as `rotate` `par` `pad` ...

**Reply by christoph-heinrich** [reply]
Isn't `vf` required for scaling down?

**Reply by hooke007** [reply]
Ah, sry I mean only `scale` was left in `vf`

**Reply by hooke007** [reply]
Comparing with the encoding step, we waste much too performance on decoding. The thumbpreview feature is almost unavailable when watching 4k-HDR sources for most users.
And I checked the recent commits, https://github.com/po5/thumbfast/commit/eb21b2e871144a328f93c4597cc1932b43783e6a seems to try to improve it.
The non-linear or non-fixed frequency of requests may greatly reduce the pressure of decoding.

**Comment by OmiChrome** [comment]
sir, lease 🙏  share  how u have that `bufffring  ` ROTATE logo  in middle?  🥺 @hooke007

**Reply by hooke007** [reply]
It's master branch.

**Reply by OmiChrome** [reply]
> It's master branch.

huh?

**Reply by hooke007** [reply]
https://github.com/tomasklaen/uosc/tree/main/scripts
